### ISSUE TITLE
Error: object eventhubs is not a member of package org.apache.spark

### ISSUE THREAD

**Issue by james-tn** [Azure/azure-event-hubs-spark#211]
I build  azure-eventhubs-spark_2.11 for version 2.1.6 but still cannot import the EventHubsUtils class
[ERROR] import org.apache.spark.eventhubs.common.EventHubsUtils

--

**Comment by sabeegrewal** [comment]
Hey James, sorry for the trouble and thanks for using the library! For that release it's:`org.apache.spark.eventhubscommon.EventHubsUtils`. There's no `.` between `eventhubs` and `common`. 

I've recently inherited this code base, so it's slightly in flux. Our next release (with Spark 2.2) will have a lot of changes. One of the most minor changes will be adding that `.` between `eventhubs` and `common`. 👍  Most of the other changes are in the wiki!

All upcoming changes will be merged into `master` leading up to the release. Check out branch `spark-2.1` for the code base used for the `azure-eventhubs-spark_2.11:2.1.6` release. 

Let me know if this works or not! Take it easy :)

**Comment by james-tn** [comment]
Thank you,
I looked that the code branch and modified my import to  
import org.apache.spark.streaming.eventhubs.EventHubsUtils
Then I could refer to the EventHubsUtils.
However, when running the example to connect spark streaming to my iot hub, this error happens within spark: 
scala>     val eventhubParameters = Map[String, String] (
     |       "eventhubs.policyname" -> policyName,
     |       "eventhubs.policykey" -> policykey,
     |       "eventhubs.namespace" -> namespace,
     |       "eventhubs.name" -> name,
     |       "eventhubs.partition.count" -> "32",
     |       "eventhubs.consumergroup" -> "$Default"
     |     )

xxxxxx

scala> val ssc = new StreamingContext(sc, Seconds(batchDuration))
ssc: org.apache.spark.streaming.StreamingContext = org.apache.spark.streaming.StreamingContext@22825762

scala> val inputDirectStream = EventHubsUtils.createDirectStreams( ssc,
     | 
     |       namespace,
     | 
     |       progressDir,
     | 
     |       Map(name -> eventhubParameters))
inputDirectStream: org.apache.spark.streaming.eventhubs.EventHubDirectDStream = org.apache.spark.streaming.eventhubs.EventHubDirectDStream@7c48ea9e

scala> 

scala> 

scala> 

scala> inputDirectStream.foreachRDD { rdd =>
     | 
     |       rdd.flatMap(eventData => new String(eventData.getBody).split(" ").map(_.replaceAll(
     | 
     |         "[^A-Za-z0-9 ]", ""))).map(word => (word, 1)).reduceByKey(_ + _).collect().toList.
     | 
     |         foreach(println)
     | 
     |     }
warning: there was one deprecation warning; re-run with -deprecation for details

scala>     ssc.start()
java.io.IOException: No FileSystem for scheme: null
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2660)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2667)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:94)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2703)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2685)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:373)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
	at org.apache.spark.streaming.eventhubs.checkpoint.DirectDStreamProgressTracker.initProgressFileDirectory(DirectDStreamProgressTracker.scala:61)
	at org.apache.spark.streaming.eventhubs.checkpoint.DirectDStreamProgressTracker.init(DirectDStreamProgressTracker.scala:116)
	at org.apache.spark.streaming.eventhubs.checkpoint.DirectDStreamProgressTracker$.initInstance(DirectDStreamProgressTracker.scala:208)
	at org.apache.spark.streaming.eventhubs.EventHubDirectDStream.start(EventHubDirectDStream.scala:132)
	at org.apache.spark.streaming.DStreamGraph$$anonfun$start$5.apply(DStreamGraph.scala:49)
	at org.apache.spark.streaming.DStreamGraph$$anonfun$start$5.apply(DStreamGraph.scala:49)
	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.foreach_quick(ParArray.scala:143)
	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.foreach(ParArray.scala:136)
	at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:972)
	at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
	at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
	at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
	at scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
	at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:969)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:152)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
	at scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
17/11/12 20:43:05 ERROR StreamingContext: Error starting the context, marking it as stopped
java.io.IOException: No FileSystem for scheme: null
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2660)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2667)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:94)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2703)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2685)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:373)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:295)
	at org.apache.spark.streaming.eventhubs.checkpoint.DirectDStreamProgressTracker.initProgressFileDirectory(DirectDStreamProgressTracker.scala:61)
	at org.apache.spark.streaming.eventhubs.checkpoint.DirectDStreamProgressTracker.init(DirectDStreamProgressTracker.scala:116)
	at org.apache.spark.streaming.eventhubs.checkpoint.DirectDStreamProgressTracker$.initInstance(DirectDStreamProgressTracker.scala:208)
	at org.apache.spark.streaming.eventhubs.EventHubDirectDStream.start(EventHubDirectDStream.scala:132)
	at org.apache.spark.streaming.DStreamGraph$$anonfun$start$5.apply(DStreamGraph.scala:49)
	at org.apache.spark.streaming.DStreamGraph$$anonfun$start$5.apply(DStreamGraph.scala:49)
	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.foreach_quick(ParArray.scala:143)
	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.foreach(ParArray.scala:136)
	at scala.collection.parallel.ParIterableLike$Foreach.leaf(ParIterableLike.scala:972)
	at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply$mcV$sp(Tasks.scala:49)
	at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
	at scala.collection.parallel.Task$$anonfun$tryLeaf$1.apply(Tasks.scala:48)
	at scala.collection.parallel.Task$class.tryLeaf(Tasks.scala:51)
	at scala.collection.parallel.ParIterableLike$Foreach.tryLeaf(ParIterableLike.scala:969)
	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask$class.compute(Tasks.scala:152)
	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:443)
	at scala.concurrent.forkjoin.RecursiveAction.exec(RecursiveAction.java:160)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
	at ... run in separate thread using org.apache.spark.util.ThreadUtils ... ()
	at org.apache.spark.streaming.StreamingContext.liftedTree1$1(StreamingContext.scala:577)
	at org.apache.spark.streaming.StreamingContext.start(StreamingContext.scala:571)
	at $line29.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(<console>:32)
	at $line29.$read$$iw$$iw$$iw$$iw$$iw$$iw$$iw.<init>(<console>:37)
	at $line29.$read$$iw$$iw$$iw$$iw$$iw$$iw.<init>(<console>:39)
	at $line29.$read$$iw$$iw$$iw$$iw$$iw.<init>(<console>:41)
	at $line29.$read$$iw$$iw$$iw$$iw.<init>(<console>:43)
	at $line29.$read$$iw$$iw$$iw.<init>(<console>:45)
	at $line29.$read$$iw$$iw.<init>(<console>:47)
	at $line29.$read$$iw.<init>(<console>:49)

**Comment by sabeegrewal** [comment]
Based on the error, it looks like you're passing in `null` as your progress directory path which isn't allowed. Also, make sure you're using the right info from your IoT Hub. We have instructions on that [here](https://github.com/Azure/azure-event-hubs-spark/blob/master/docs/connect_iothub_spark.md).

**Comment by james-tn** [comment]
Wonderful! I am now able to run spark streaming on IoT Hub messages.
Indeed I had both problem, wrong directory name and wrong IoT hub settings.
(I think the page on https://docs.microsoft.com/en-us/azure/hdinsight/spark/apache-spark-eventhub-streaming is outdated)
Will need to catch up on Scala as I've only worked with spark streaming with python.
Hope that the library for python will also be available in near future.
Thank you so much for your prompt support.

**Comment by sabeegrewal** [comment]
No worries, glad it's working! I'll go through and update that doc tomorrow. 

I have to wait on the Python EventHubs client to be finished, but we've definitely had a few requests for Python support. I'll look into that more seriously during Q1/Q2 of next  year. 

Also, as you're brushing off your scala, feel free to join the [gitter](https://gitter.im/azure-event-hubs-spark/Lobby?source=orgpage), it's a good place to chat about the library....I'm there quite a lot!
